// discord-formatter/dist/discord-formatter.umd.js
(function (factory) {
    typeof define === 'function' && define.amd ? define(factory) :
    factory();
}((function () { 'use strict';

    Object.defineProperty(exports, "__esModule", { value: true });
    exports.formatChannel = void 0;
    function formatChannel(id) {
        return `<#${id}>`;
    }
    exports.formatChannel = formatChannel;

    Object.defineProperty(exports, "__esModule", { value: true });
    exports.formatEmoji = void 0;
    function formatEmoji(name, id, animated = false) {
        return `<${animated ? 'a' : ''}:${name}:${id}>`;
    }
    exports.formatEmoji = formatEmoji;

    Object.defineProperty(exports, "__esModule", { value: true });
    exports.formatRole = void 0;
    function formatRole(id) {
        return `<@&${id}>`;
    }
    exports.formatRole = formatRole;

    Object.defineProperty(exports, "__esModule", { value: true });
    exports.formatTimestamp = exports.TimestampStyle = void 0;
    var TimestampStyle;
    (function (TimestampStyle) {
        TimestampStyle["ShortTime"] = "SHORT_TIME";
        TimestampStyle["LongTime"] = "LONG_TIME";
        TimestampStyle["ShortDate"] = "SHORT_DATE";
        TimestampStyle["LongDate"] = "LONG_DATE";
        TimestampStyle["ShortDateTime"] = "SHORT_DATE_TIME";
        TimestampStyle["LongDateTime"] = "LONG_DATE_TIME";
        TimestampStyle["RelativeTime"] = "RELATIVE_TIME";
    })(TimestampStyle = exports.TimestampStyle || (exports.TimestampStyle = {}));
    const styles = {
        [TimestampStyle.ShortTime]: 't',
        [TimestampStyle.LongTime]: 'T',
        [TimestampStyle.ShortDate]: 'd',
        [TimestampStyle.LongDate]: 'D',
        [TimestampStyle.ShortDateTime]: 'f',
        [TimestampStyle.LongDateTime]: 'F',
        [TimestampStyle.RelativeTime]: 'R'
    };
    function formatTimestamp(date, style = TimestampStyle.ShortDateTime) {
        return `<t:${date instanceof Date ? Math.floor(date.getTime() / 1000) : date}:${styles[style]}>`;
    }
    exports.formatTimestamp = formatTimestamp;

    Object.defineProperty(exports, "__esModule", { value: true });
    exports.formatUser = void 0;
    function formatUser(id, hasNickname = false) {
        return `<@${hasNickname ? '!' : ''}${id}>`;
    }
    exports.formatUser = formatUser;

    Object.defineProperty(exports, "__esModule", { value: true });

})));
