// discord-formatter/dist/formatTimestamp.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.formatTimestamp = exports.TimestampStyle = void 0;
var TimestampStyle;
(function (TimestampStyle) {
    TimestampStyle["ShortTime"] = "SHORT_TIME";
    TimestampStyle["LongTime"] = "LONG_TIME";
    TimestampStyle["ShortDate"] = "SHORT_DATE";
    TimestampStyle["LongDate"] = "LONG_DATE";
    TimestampStyle["ShortDateTime"] = "SHORT_DATE_TIME";
    TimestampStyle["LongDateTime"] = "LONG_DATE_TIME";
    TimestampStyle["RelativeTime"] = "RELATIVE_TIME";
})(TimestampStyle = exports.TimestampStyle || (exports.TimestampStyle = {}));
const styles = {
    [TimestampStyle.ShortTime]: 't',
    [TimestampStyle.LongTime]: 'T',
    [TimestampStyle.ShortDate]: 'd',
    [TimestampStyle.LongDate]: 'D',
    [TimestampStyle.ShortDateTime]: 'f',
    [TimestampStyle.LongDateTime]: 'F',
    [TimestampStyle.RelativeTime]: 'R'
};
function formatTimestamp(date, style = TimestampStyle.ShortDateTime) {
    return `<t:${date instanceof Date ? Math.floor(date.getTime() / 1000) : date}:${styles[style]}>`;
}
exports.formatTimestamp = formatTimestamp;
